File: src/test/console/java/src/test/java/kobiton/com/utils/AutomationUtils.java
package kobiton.com.utils;

import com.kobiton.config.Config;
import com.kobiton.service.KeyService;
import com.kobiton.service.model.APIKey;
import com.kobiton.service.model.Device;
import java.io.IOException;
import java.net.MalformedURLException;
import java.net.URL;
import java.util.logging.Level;
import java.util.logging.Logger;
import org.openqa.selenium.remote.DesiredCapabilities;

public class AutomationUtils {    
    public static final URL kobitonServerUrl() {             
        try {
            APIKey key = new KeyService().getApiKey();
            
            URL apiUrl = new URL(Config.getInstance().getApiUrl());
            String automationUrl = String.format("%s://%s:%s@%s/wd/hub", 
                    apiUrl.getProtocol(),
                    key.getUsername(),
                    key.getKey(),
                    apiUrl.getHost()
            );
            Logger.getLogger(AutomationUtils.class.getName()).log(Level.SEVERE, "automationUrl:" + automationUrl);
            return new URL(automationUrl);
        } catch (MalformedURLException e) {
            Logger.getLogger(AutomationUtils.class.getName()).log(Level.SEVERE, null, e);
        } catch (IOException ex) {
            Logger.getLogger(AutomationUtils.class.getName()).log(Level.SEVERE, null, ex);
        }
        return null;
    }

    public static final DesiredCapabilities desiredCapabilitiesAndroidWeb(Device device) {
        DesiredCapabilities capabilities = createCapabilitiesFor(device);
        capabilities.setCapability("sessionName", "Android Web");
        capabilities.setCapability("sessionDescription", "This is an example for Android Web testing");
        capabilities.setCapability("browserName", "chrome");
        return capabilities;
    }

    public static final DesiredCapabilities desiredCapabilitiesAndroidApp(Device device) {
        DesiredCapabilities capabilities = createCapabilitiesFor(device);
        capabilities.setCapability("sessionName", "Android app");
        capabilities.setCapability("sessionDescription", "This is an example for Android app testing");
        capabilities.setCapability("app", "https://s3.amazonaws.com/kobiton-dev/apps-test/demo/com.dozuki.ifixit.apk");
        return capabilities;
    }

    public static final DesiredCapabilities desiredCapabilitiesiOSWeb(Device device) {
        DesiredCapabilities capabilities = createCapabilitiesFor(device);
        capabilities.setCapability("sessionName", "iOS Web");
        capabilities.setCapability("sessionDescription", "This is an example for iOS Web testing");
        capabilities.setCapability("browserName", "safari");
        return capabilities;
    }

    public static final DesiredCapabilities desiredCapabilitiesiOSApp(Device device) {
        DesiredCapabilities capabilities = createCapabilitiesFor(device);
        capabilities.setCapability("sessionName", "iOS app");
        capabilities.setCapability("sessionDescription", "This is an example for iOS App testing");
        capabilities.setCapability("app", "https://s3.amazonaws.com/kobiton-dev/apps-test/demo/iFixit.ipa");
        return capabilities;
    }
    
    private static DesiredCapabilities createCapabilitiesFor(Device device) {
        DesiredCapabilities capabilities = new DesiredCapabilities();
        capabilities.setCapability("deviceOrientation", "portrait");
        capabilities.setCapability("captureScreenshots", true);
        capabilities.setCapability("deviceGroup", "KOBITON");
        capabilities.setCapability("deviceName", device.getDeviceName());
        capabilities.setCapability("platformName", device.getPlatformName());
        capabilities.setCapability("platformVersion", device.getPlatformVersion());
        return capabilities;
    }
}

File: src/test/console/dotnet/MobileAutomation/Tests/Utils/AutomationUtils.cs
using System;
using MobileAutomation.Config;
using MobileAutomation.Service;
using MobileAutomation.Service.Model;
using OpenQA.Selenium.Remote;

namespace MobileAutomation.Tests.Utils
{
  public class AutomationUtils
  {
    private static readonly AutomationUtils instance = new AutomationUtils();
    public static AutomationUtils Instance
    {
        get
        {
            return instance;
        }
    }

    public Uri KobitonServerUrl { get; internal set; }

    private AutomationUtils()
    {
      KobitonServerUrl = new Uri("https://api.kobiton.com/wd/hub");
    }

    public DesiredCapabilities DesiredCapabilitiesAndroidWeb(Device device)
    {
      var capabilities = CreateCapabilitiesFor(device);
      capabilities.SetCapability("sessionName", "[C#]Android Web");
      capabilities.SetCapability("sessionDescription", "This is an example for Android Web testing");
      capabilities.SetCapability("browserName", "chrome");
      return capabilities;
    }

    public DesiredCapabilities DesiredCapabilitiesAndroidApp(Device device)
    {
      var capabilities = CreateCapabilitiesFor(device);
      capabilities.SetCapability("sessionName", "[C#]Android app");
      capabilities.SetCapability("sessionDescription", "This is an example for Android app testing");
      capabilities.SetCapability("app", "https://s3-ap-southeast-1.amazonaws.com/kobiton-devvn/apps-test/demo/com.dozuki.ifixit.apk");
      return capabilities;
    }

    public DesiredCapabilities DesiredCapabilitiesIOSWeb(Device device)
    {
      var capabilities = CreateCapabilitiesFor(device);
      capabilities.SetCapability("sessionName", "[C#]iOS web");
      capabilities.SetCapability("sessionDescription", "This is an example for iOS web testing");
      capabilities.SetCapability("browserName", "safari");
      return capabilities;
    }

    public DesiredCapabilities DesiredCapabilitiesIOSApp(Device device)
    {
      var capabilities = CreateCapabilitiesFor(device);
      capabilities.SetCapability("sessionName", "[C#]iOS app");
      capabilities.SetCapability("sessionDescription", "This is an example for iOS app testing");
      capabilities.SetCapability("app", "https://s3-ap-southeast-1.amazonaws.com/kobiton-devvn/apps-test/demo/iFixit.ipa");
      return capabilities;
    }

    private DesiredCapabilities CreateCapabilitiesFor(Device device)
    {
      var capabilities = new DesiredCapabilities();
      var keyService = new KeyService();
      var key = keyService.GetAPIKey();
      capabilities.SetCapability("username", key.Username);
      capabilities.SetCapability("accessKey", key.Key);
      capabilities.SetCapability("deviceOrientation", "portrait");
      capabilities.SetCapability("captureScreenshots", true);
      capabilities.SetCapability("deviceGroup", "KOBITON");
      capabilities.SetCapability("deviceName", device.DeviceName);
      capabilities.SetCapability("platformName", device.PlatformName);
      capabilities.SetCapability("platformVersion", device.PlatformVersion);
      return capabilities;
    }
  }
}
